misc-tools/balloons.in

```diff
@@ -23,51 +36,128 @@ define('LOGFILE', LOGDIR.'/balloons.log');
 
 require(LIBDIR . '/init.php');
 
-setup_database_connection();
-
 $verbose = LOG_INFO;
 
-$waittime = 5;
+define('WAITTIME', 5);
+
+function usage() : void
+{
+    echo "Usage: " . SCRIPT_ID . " API_LOCATION\n" .
+        "Send out notifications on each new balloon.\n\n" .
+        "This script is not needed when using the web interface to handle balloons.\n\n" .
+        "API_LOCATION is the base of your DOMjudge installation, without the\n" .
+        "'/api' auffix and no trailing slash, e.g. https://example.org/domjudge\n\n" .
+        "Credentials for this API need to be in a .netrc file in your home directory,\n" .
+        "please refer to the DOMjudge manual for the format.\n\n";
+        "The way to notify is configured inside this script by setting BALLOON_CMD.\n\n";
+    exit;
+}
 
 /**
  * Returns a text to be sent when notifying of a new balloon.
  */
-function notification_text($team, $problem, $contest, $probs_solved, $probs_data, $comment)
+function notification_text(array $contest, string $teamname, string $location = null, string $problem, string $color, array $probs_solved, string $comment = null) : string
```

Comment:
I think having optional parameters (i.e. string `$location = null`) before non optional ones is not officially allowed. We can change it to `?string $location` to denote it can be optional at least, otherwise you can not even provide a non-null value.